target/linux/ipq40xx/base-files/lib/preinit/05_set_iface_mac_ipq40xx.sh

```diff
@@ -22,6 +22,12 @@ preinit_set_mac_address() {
 		mac_lan=$(get_mac_binary "/sys/bus/i2c/devices/0-0050/eeprom" 0x66)
 		[ -n "$mac_lan" ] && ip link set dev eth0 address "$mac_lan"
 		;;
+	xiaomi,meshd01)
+		ip link set dev eth0 address $(mtd_get_mac_binary "0:ART" 0x0)
+		ip link set dev eth1 address $(mtd_get_mac_binary "0:ART" 0x6)
+		ip link set dev eth2 address $(mtd_get_mac_binary "0:ART" 0xC)
```

Comment:
I'd prefer lower-case: 0xc.
Have you checked that the assignment corresponds to stock firmware?

Reply:
yes, stock firmware extract this same way using dump of ART and hexdump :)

`
get_eth0()
{
	eth0adr="`hexdump -s 0 -n 6 -e '6/1 "%02x:"'  /dev/"$art_mtd" | cut -b-17`"
}

get_eth1()
{
	eth1adr="`hexdump -s 6 -n 6 -e '6/1 "%02x:"'  /dev/"$art_mtd" | cut -b-17`"
}

get_eth2()
{
	eth2adr="`hexdump -s 12 -n 6 -e '6/1 "%02x:"'  /dev/"$art_mtd" | cut -b-17`"
}

get_eth3()
{
	eth3adr="`hexdump -s 18 -n 6 -e '6/1 "%02x:"'  /dev/"$art_mtd" | cut -b-17`"
}
`